storage/tianmu/core/query_compile.cpp

```diff
@@ -1078,6 +1078,7 @@ int Query::Compile(CompiledQuery *compiled_query, SELECT_LEX *selects_list, SELE
       if (!AddOrderByFields(order, tmp_table, group != nullptr || sl->join->select_distinct || aggr_used))
         throw CompilationError();
       CondID cond_id;
+      internalDataTimeIsNullToEq(sl->master_unit()->thd, conds, &conds);

```

Comment:
if you want to do the type test, i think this test should be in `BuildConditions`.